ophyd/signal.py

```diff
@@ -1165,13 +1173,6 @@ def describe(self):
 
         return {self.name: desc}
 
-    def destroy(self):
-        '''Disconnect the EpicsSignal from the underlying PV instance'''
-        super().destroy()
-        if self._read_pv is not None:
-            self.cl.release_pvs(self._read_pv)
-            self._read_pv = None
```

Comment:
Short version: Do you think there's any value in keeping the `self._read_pv = None` in the new `finalize` for potential side-effects? (and same for the write version of course)

Long version:

This follows suit with the example in the weakref docs, removing the `not None` check and `= None` on `_read_pv`. Same with the write. The idea being that finalizers can only be invoked once, so there's no need to guard from repeated calls.

But the weakref example is very simple; their `self.name` is just a string and setting it to `None` served no other purpose than that condition. Here, `self._read_pv` is used in many other places and is a PV object. Setting it to `None` (or not) may have other consequences (including when the PV object is garbage collected).

That said, none of the tests are failing and I can't think of any real scenario where this would make a tangible difference. I think it would only come into play if `destroy` was called manually, and the reference to the `Signal` instance kept around (or even attempted to be used again, which you shouldn't do anyway). So feel free to handwave this as a non-issue.

Reply:
I'm of two minds about this.

A "dead" object is more useful for post-mortem debugging than `None`. If the state is a file, one often keeps a closed file handle, which can still be inspected to show its name as so on. Here we might keep a released PV for the same reason.

I usually err on the side of cleaning up state rather than keeping stuff around "in case you need it" for debugging, particularly in the case of large data structures that take up significant memory, but if the released PV is no longer using network resources of significant memory, I'm inclined to keep it for convenience if we cannot think of any bad implications.

Reply:
Sounds reasonable to me, and I can't think of any downside in practice.